datetime.fifthtry.site/js/functions.js

```diff
@@ -2,29 +2,64 @@ function datetime(diff) {
     const currentDate = new Date();
     const date = new Date(currentDate.getTime() + diff * 60000);
 
-    const utcTime = new Date(
-        date.getUTCFullYear(),
-        date.getUTCMonth(),
-        date.getUTCDate(),
-        date.getUTCHours(),
-        date.getUTCMinutes(),
-        date.getUTCSeconds(),
-        date.getUTCMilliseconds()
-    );
+    // Date part: YYYYMMDD (always 8 digits)
+    const year = date.getUTCFullYear();
+    const month = (date.getUTCMonth() + 1).toString().padStart(2, "0");
+    const day = date.getUTCDate().toString().padStart(2, "0");
+    const datePart = parseInt(`${year}${month}${day}`);
+
+    // Time part: HHMMSSmmmnnnnnnnnn (always 17 digits, zero-padded)
+    const hours = date.getUTCHours().toString().padStart(2, "0");
+    const minutes = date.getUTCMinutes().toString().padStart(2, "0");
+    const seconds = date.getUTCSeconds().toString().padStart(2, "0");
+    const milliseconds = date.getUTCMilliseconds().toString().padStart(3, "0");
+    // JS Date does not provide nanosecond precision, so pad with zeros
+    const nanoseconds = "000000000";
+    const timeStr = `${hours}${minutes}${seconds}${milliseconds}${nanoseconds}`;
+    // Store as string to preserve leading zeros
 
-    const now = BigInt(utcTime) * BigInt(1000000);
     return new fastn.recordInstanceClass({
-        dt: now,
+        date: datePart,
+        time: timeStr,
     });
 }
 
+function constructSingleInt(dt) {
```

Comment:
I would try to sync the function name in JS with the corresponding function name in ftd. It is not great calling the same thing `combine` in one place, and `constructSingleInt` in another. Both can be `combine`. 

Having said that, I prefer `to-unixtime`.